go/worker/client/committee/node.go

```diff
@@ -89,8 +91,15 @@ func (n *Node) HandleNewBlockLocked(*runtime.BlockInfo) {
 }
 
 // HandleRuntimeHostEventLocked is guarded by CrossNode.
-func (n *Node) HandleRuntimeHostEventLocked(*host.Event) {
-	// Nothing to do here.
+func (n *Node) HandleRuntimeHostEventLocked(ev *host.Event) {
+	if n.roleProvider != nil {
```

Comment:
```suggestion
	if n.roleProvider == nil {
		return
	}
```
And outdent the below.